app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TweetsController < ApplicationController
+  
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    if  @user
+      erb :'tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find(session[:id])
```

Comment:
Looks like you're missing a `current_user` helper method in your application controller. It is preferable to use that method both because it is more expressive and can save some session/database lookups through Memoization ([read more](http://gavinmiller.io/2013/basics-of-ruby-memoization/) and [here](http://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/)).